src/js/geo-dash.js

```diff
@@ -75,23 +76,68 @@ class Geodash extends React.Component {
+            .catch(response => {
+                console.log(response);
+                alert("Error initializing Geo-Dash. See console for details.");
+            });
     }
 
+    getInstitutionImagery = () => fetch(`/get-institution-imagery?institutionId=${this.state.institutionId}`)
+        .then(response => response.ok ? response.json() : Promise.reject(response))
+        .then(data => this.setState({imageryList: data}));
+
+    getWidgetsByProjectId = () => fetch(`/geo-dash/get-by-projid?projectId=${this.state.projectId}`)
+        .then(response => response.ok ? response.json() : Promise.reject(response))
+        .then(data => data.widgets.map(widget => {
+            widget.isFull = false;
+            widget.opacity = 0.9;
+            widget.sliderType = widget.swipeAsDefault ? "swipe" : "opacity";
+            widget.swipeValue = 1.0;
+            return widget;
+        }));
+
+    getVectorSource = () => {
+        const plotShape = this.getParameterByName("plotShape");
+        const radius = this.getParameterByName("bradius");
+        const center = this.getParameterByName("bcenter");
+        const plotId = this.getParameterByName("plotId");
+        if (plotShape === "polygon") {
+            return fetch(`/get-plot-sample-geom?plotId=${plotId}`)
+                .then(response => response.ok ? response.json() : Promise.reject(response))
+                .then(plotJsonObject => {
+                    const sampleGeom = (plotJsonObject.samples || []).map(e => e.geom);
+                    const features = [plotJsonObject.geom].concat(sampleGeom)
+                        .filter(e => e)
+                        .map(geom => new Feature({geometry: mercator.parseGeoJson(geom, true)}));
+                    return new Vector({features: features});
+                });
+        } else if (plotShape === "square") {
+            const pointFeature = new Feature(
+                new Point(projTransform(JSON.parse(center).coordinates, "EPSG:4326", "EPSG:3857"))
+            );
+            const pointExtent = pointFeature.getGeometry().getExtent();
+            const bufferedExtent = new ExtentBuffer(pointExtent, parseInt(radius));
+            return Promise.resolve(new Vector({
+                features: [new Feature(new Polygon(
+                    [[[bufferedExtent[0], bufferedExtent[1]],
+                      [bufferedExtent[0], bufferedExtent[3]],
+                      [bufferedExtent[2], bufferedExtent[3]],
+                      [bufferedExtent[2], bufferedExtent[1]],
+                      [bufferedExtent[0], bufferedExtent[1]]]]
+                )
+                )],
+            }));
+        } else if (plotShape === "circle") {
+            return Promise.resolve(new Vector({
+                features: [new Feature(new Circle(
+                    projTransform(JSON.parse(center).coordinates, "EPSG:4326", "EPSG:3857"), radius * 1)
+                )],
+            }));
+        }
```

Comment:
what does this function return if plotshape is not polygon, circle or square?

Reply:
Is there another option for the plot shape?